tests/test_apply.py

```diff
@@ -1212,3 +1213,20 @@ def test_pauliz_hadamard(self, theta, phi, varphi, monkeypatch, qubit_device_3_w
             - 2 * np.cos(theta) * np.sin(phi) * np.sin(2 * varphi)
         ) / 4
         assert np.allclose(var, expected, atol=tol, rtol=0)
+
+
+@pytest.mark.parametrize("wires", range(2, 10))
+def test_qubit_unitary(wires):
+    """Test if lightning.qubit successful applies an arbitrary unitary"""
+
+    unitary = unitary_group.rvs(2 ** wires, random_state=0)
+
+    dev = qml.device("lightning.qubit", wires=wires)
+
+    @qml.qnode(dev)
+    def f():
+        qml.QubitUnitary(unitary, wires=range(wires))
+        return qml.state()
+
+    target_state = unitary[:, 0]
+    assert np.allclose(f(), target_state)
```

Comment:
Variants of this test are almost certainly provided by the device test suite, but I'm happy to have it explicitly here even if there is some redundancy.